raiden/storage/sqlite.py

```diff
@@ -292,6 +307,89 @@ def get_latest_state_change_by_data_field(
 
         return result
 
+    def _get_state_changes(
+            self,
+            limit: int = None,
+            offset: int = None,
+            filters: Dict[str, Any] = None,
+    ) -> List[StateChangeRecord]:
+        """ Return a batch of state change records (identifier and data)
+
+        The batch size can be tweaked with the `limit` and `offset` arguments.
+
+        Additionally the returned state changes can be optionally filtered with
+        the `filters` parameter to search for specific data in the state change data.
+        """
+        limit, offset = _sanitize_limit_and_offset(limit, offset)
+        cursor = self.conn.cursor()
+
+        query = 'SELECT identifier, data FROM state_changes '
+
+        where_clauses = []
+        args = []
+        if filters:
+            for field, value in filters.items():
+                where_clauses.append('json_extract(data, ?)=?')
+                args.append(f'$.{field}')
+                args.append(value)
+
+            query += (
+                f"WHERE {' AND '.join(where_clauses)}"
+            )
+
+        query += 'ORDER BY identifier ASC LIMIT ? OFFSET ?'
+        args.append(limit)
+        args.append(offset)
+
+        cursor.execute(query, args)
+        result = []
+        try:
+            rows = cursor.fetchall()
+            for row in rows:
+                result.append(StateChangeRecord(
+                    state_change_identifier=row[0],
+                    data=row[1],
+                ))
+        except AttributeError:
+            raise InvalidDBData(
+                'Your local database is corrupt. Bailing ...',
+            )
+
+        return result
+
+    def batch_query_state_changes(
+            self,
+            batch_size: int,
+            filters: Dict[str, Any] = None,
+    ) -> List[StateChangeRecord]:
+        """Batch query state change records with a given batch size and an optional filter
+
+        This is a generator function returning each batch to the caller to work with.
+        """
+        limit = batch_size
+        offset = 0
+        result_length = 1
+
+        while result_length != 0:
+            result = self._get_state_changes(
+                limit=limit,
+                offset=offset,
+                filters=filters,
+            )
+            result_length = len(result)
+            offset += result_length
+            yield result
+
+    def update_state_changes(self, state_changes: List[StateChangeRecord]) -> None:
+        """Given a list of identifier/data state change records update them in the DB"""
+        cursor = self.conn.cursor()
+        for state_change in state_changes:
```

Comment:
why don't you use executemany here too?

Reply:
Done in https://github.com/raiden-network/raiden/pull/3550/commits/0aed2b6b9217d6c6324608e2d20d78e9d00f762a